singd/structures/recursive.py

```diff
@@ -184,6 +184,17 @@ def __init__(self, A: StructuredMatrix, B: Tensor, C: StructuredMatrix):
         self.C: StructuredMatrix
         self.register_substructure(C, "C")
 
+    @property
+    def shape(self) -> Size:
```

Comment:
I think unfortunately no, because we don't know if the recursion occurs on a diagonal or off-diagonal block, and the class does not allow to detect if a structure sits on the diagonal or not. Otherwise, we could indeed implement this by summing the dimensions over all structures on the diagonal.

Reply:
But looking at the file again, I found some code that I had forgotten to refactor, thanks for the closer look!